### ISSUE TITLE
Perl::Critic::Policy::Freenode::DeprecatedFeatures doesn't understand postfix deref

### ISSUE THREAD

**Issue by simcop2387** [Grinnz/Perl-Critic-Freenode#26]
The following snippet results in a perlcritic issue

```perl
use strict;
use warnings;

my $foo = {workers => {a => b => c =>4}};

if (keys $foo->workers->%* < 4) {
  print "Warnings";
}
```

autoderef is deprecated at line 6, column 5.  Use of each/keys/pop/push/shift/splice/unshift/values on a reference is an experimental feature that is removed in perl v5.24.0. Dereference the array or hash to use these functions on it.  (Severity: 4)

**Comment by Grinnz** [comment]
Thanks. I'll take a stab at recognizing this case but I'm not sure if PPI recognizes this syntax yet either.

**Comment by Grinnz** [comment]
Actually I am already checking for this case but my hack got fooled by the < 4 at the end.

**Comment by Grinnz** [comment]
Fixed in 0.021
